### PR TITLE
Add Typescript support

### PR DESCRIPTION
**Pull request by frederickfogerty**
This PR supersedes the work done by @Ephys in #51.

This PR:

- fixes the type declaration (most of the work was done by @Ephys)
- adds automated tests with `typescript-definition-tester` to make sure the typescript declaration compile correctly against example code. See `examples/typescripts/index.tsx`
- adds some types required for the typescript to `devDependencies`
- adds some documentation about using typescript.

### CODE REVIEW

**Comment by philcockfield** [comment]
Great work @frederickfogerty !!!  I've been hanging out for this.

**Comment by frederickfogerty** [comment]
Okay, this time @arunoda @roonyh I didn't break the build, it's a problem with existing packages.

**Comment by arunoda** [comment]
This looks pretty cool to me.
@roonyh Let's take this.

**Comment by roonyh** [comment]
Published version 1.4.0 with this. Thanks @frederickfogerty and @Ephys 👍

File: storybook-addon-knobs.d.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+export function boolean(name: string, value: boolean): boolean;
+
+export function number(name: string, value: number): number;
+
+export function object(name: string, value: Object): Object;
+
+export function select<T>(name: string, options: { [s: string]: T }, value: string): T;
+export function select(name: string, options: string[], value: string): string;
+
+export function date(name: string, value?: Date);
```
Comment (aivarsak) [inline storybook-addon-knobs.d.ts]: No return type for date?

Comment (frederickfogerty) [inline storybook-addon-knobs.d.ts]: This one has obviously slipped through the cracks

Comment (frederickfogerty) [inline storybook-addon-knobs.d.ts]: @aivarsak fixed in #67

**Pull request merged**